modules/apps/portal-vulcan/portal-vulcan-impl/src/main/java/com/liferay/portal/vulcan/internal/jaxrs/writer/interceptor/EntityExtensionWriterInterceptor.java

```diff
@@ -0,0 +1,73 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.portal.vulcan.internal.jaxrs.writer.interceptor;
+
+import com.liferay.portal.vulcan.internal.jaxrs.extension.ExtendedEntity;
+import com.liferay.portal.vulcan.jaxrs.context.ExtensionContext;
+
+import java.io.IOException;
+
+import java.util.Optional;
+
+import javax.ws.rs.core.Context;
+import javax.ws.rs.ext.Provider;
+import javax.ws.rs.ext.Providers;
+import javax.ws.rs.ext.WriterInterceptor;
+import javax.ws.rs.ext.WriterInterceptorContext;
+
+/**
+ * @author Javier de Arcos
+ */
+@Provider
+public class EntityExtensionWriterInterceptor implements WriterInterceptor {
+
+	public EntityExtensionWriterInterceptor(@Context Providers providers) {
```

Comment:
I changed the interceptors to make the providers injection in the constructor, because it was easier for testing them injecting a mock, but then it didn't work at runtime... Something obvious I'm missing?

Tomorrow I will go back to inject providers at field level and test using Powermock or something like that

Reply:
Mmmm, I don't know if all types (Providers, QueryParams) are supported at the field level.